components/builder-protocol/src/vault.rs

```diff
@@ -284,9 +284,16 @@ impl Persistable for Project {
 impl ToJson for Project {
     fn to_json(&self) -> Json {
         let mut m = BTreeMap::new();
+        let mut v = BTreeMap::new();
         m.insert("id".to_string(), self.get_id().to_json());
         m.insert("plan_path".to_string(),
                  self.get_plan_path().to_string().to_json());
+
+        v.insert("type".to_string(), "git".to_string().to_json());
```

Comment:
You might considering implementing `ToJson for VCSGit` and then calling `m.insert("vcs".to_string(), self.get_git().to_json()`